src/views/common/Animation/AnimationControls/PlaybackControlButtonLogic.ts

```diff
@@ -0,0 +1,80 @@
+
+export type PlaybackButtonLogicCallback = (e: React.MouseEvent<any> | React.KeyboardEvent<HTMLDivElement>) => void
+export type KeyHandler = (e: React.KeyboardEvent<HTMLDivElement>) => void
+
+export type PlaybackControlLogic = {
+    returnToBeginningHandler: PlaybackButtonLogicCallback
+    backSkipHandler: PlaybackButtonLogicCallback
+    forwardSkipHandler: PlaybackButtonLogicCallback
+    jumpToEndHandler: PlaybackButtonLogicCallback
+    playPauseHandler: PlaybackButtonLogicCallback
+    reverseDirectionHandler: PlaybackButtonLogicCallback
+    syncWindowHandler: PlaybackButtonLogicCallback
+    cropWindowHandler: PlaybackButtonLogicCallback
+    keyboardControlHandler: KeyHandler
+}
+
+const usePlaybackControlButtonLogic = (dispatch: AnimationStateDispatcher<any>): PlaybackControlLogic => {
+    const handleSkipArrow = useCallback((mode: 'end' | 'skip', backward?: boolean) => {
+        return (e: React.MouseEvent | React.KeyboardEvent) => {
+            dispatch({
+                type: mode === 'end' ? 'TO_END' : 'SKIP',
+                backward: backward
+            })
+        }
+    }, [dispatch])
+
+    const returnToBeginningHandler = useMemo(() => handleSkipArrow('end', true), [handleSkipArrow])
+    const backSkipHandler = useMemo(() => handleSkipArrow('skip', true), [handleSkipArrow])
+    const forwardSkipHandler = useMemo(() => handleSkipArrow('skip'), [handleSkipArrow])
+    const jumpToEndHandler = useMemo(() => handleSkipArrow('end'), [handleSkipArrow])
+
+    const playPauseHandler = useCallback((e: React.MouseEvent | React.KeyboardEvent) => {
+        dispatch({type: 'TOGGLE_PLAYBACK'})
+    }, [dispatch])
+
+    const reverseDirectionHandler = useCallback((e: React.MouseEvent | React.KeyboardEvent) => {
+        dispatch({type: 'REVERSE_REPLAY_RATE'})
+    }, [dispatch])
+
+    const syncWindowHandler = useCallback((e: React.MouseEvent | React.KeyboardEvent) => {
+        dispatch({type: 'TOGGLE_WINDOW_SYNC'})
+    }, [dispatch])
+
+    const cropWindowHandler = useCallback((e: React.MouseEvent | React.KeyboardEvent) => {
+        dispatch({type: 'COMMIT_WINDOW'})
+    }, [dispatch])
+
+    const fineAdjustmentHandler = useCallback((e: React.KeyboardEvent) => {
+        // This may be ill-advised
+        if (e.key !== 'ArrowLeft' && e.key !== 'ArrowRight') return
+        dispatch({
+            type: 'SKIP',
+            backward: e.key === 'ArrowLeft',
+            fineSteps: 1,
+            frameByFrame: e.ctrlKey
+        })
+    }, [dispatch])
+
+    const keyboardControlHandler = useCallback((e: React.KeyboardEvent) => {
+        switch(e.key) {
+            case 'ArrowLeft':
+            case 'ArrowRight':
+                fineAdjustmentHandler(e)
+                break
+            case ' ':
+                playPauseHandler(e)
+                break
+            default:
+                return
+        }
+    }, [fineAdjustmentHandler, playPauseHandler])
```

Comment:
The keyboard event handler is defined here, rather than attached to individual buttons. This has pros and cons: it might be desirable to encapsulate everything about a button a little more (by having buttons define their own keystroke), but this prevents conflicting button definitions, and also makes it easier to have one central function to attach to the `FrameAnimation` object. Also, note that some keyboard-based interactions may not have a specific button to attach to--for instance the mouse-wheel controls for fine frame adjustment aren't a clickable button, and really operate on a different paradigm from buttons anyway.